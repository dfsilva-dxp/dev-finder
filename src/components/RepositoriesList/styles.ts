import styled, { css } from "styled-components";

export const List = styled.div`
  ${({ theme }) => css`
    max-height: 75vh;
    overflow: hidden;

    table {
      width: 100%;
      border-spacing: 0 0.5rem;

      th {
        padding: 1rem 2rem;
        font-weight: ${theme.font.weight.medium};
        font-size: ${theme.font.sizes.small};
        line-height: 1.5rem;
        text-align: left;
        border-radius: ${theme.border.radius};
      }

      td {
        background: ${theme.colors.panel};
        padding: 1rem 2rem;
        border: 0;

        &:last-child {
          max-width: 40ch;
          overflow: hidden;
          text-overflow: ellipsis;
          white-space: nowrap;
        }
      }
    }

    &:hover {
      overflow-y: scroll;
    }
  `}
`;
