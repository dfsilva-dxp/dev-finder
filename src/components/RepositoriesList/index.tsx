import { List } from "./styles";

export default function RepositoriesList() {
  return (
    <List>
      <table>
        <thead>
          <tr>
            <th>Name</th>
            <th>Description</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>ReactJS</td>
            <td>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Velit
              provident culpa doloribus obcaecati nam in fugit tempora ex
              numquam. Placeat fugiat odit quia qui, ex excepturi eius ut non
              sed.
            </td>
          </tr>
        </tbody>
      </table>
    </List>
  );
}
