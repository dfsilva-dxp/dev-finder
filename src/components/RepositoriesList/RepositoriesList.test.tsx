import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/test/helpers";

import RepositoriesList from ".";

describe("<RepositoriesList/>", () => {
  it("should render two column headers with the Name and Description texts", () => {
    renderWithTheme(<RepositoriesList />);

    expect(screen.getAllByRole("columnheader")).toHaveLength(2);

    expect(
      screen.getByRole("columnheader", { name: /name/i })
    ).toBeInTheDocument();

    expect(
      screen.getByRole("columnheader", { name: /description/i })
    ).toBeInTheDocument();
  });

  it("should render with currectly styles", () => {
    const { container } = renderWithTheme(<RepositoriesList />);

    expect(screen.getByRole("columnheader", { name: /name/i })).toHaveStyle({
      padding: "1rem 2rem",
      "font-weight": "500",
      "font-size": "0.875rem",
      "line-height": "1.5rem",
      "text-align": "left",
      "border-radius": "0.25rem"
    });

    expect(screen.getByRole("cell", { name: /reactjs/i })).toHaveStyle({
      background: "#393656",
      padding: "1rem 2rem",
      border: "0"
    });

    expect(container.parentNode).toMatchSnapshot();
  });
});
