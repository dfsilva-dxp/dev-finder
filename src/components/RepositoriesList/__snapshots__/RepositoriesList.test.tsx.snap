// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoriesList/> should render with currectly styles 1`] = `
.c0 {
  max-height: 75vh;
  overflow: hidden;
}

.c0 table {
  width: 100%;
  border-spacing: 0 0.5rem;
}

.c0 table th {
  padding: 1rem 2rem;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.5rem;
  text-align: left;
  border-radius: 0.25rem;
}

.c0 table td {
  background: #393656;
  padding: 1rem 2rem;
  border: 0;
}

.c0 table td:last-child {
  max-width: 40ch;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c0:hover {
  overflow-y: scroll;
}

<body>
  <div>
    <div
      class="c0"
    >
      <table>
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Description
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              ReactJS
            </td>
            <td>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Velit provident culpa doloribus obcaecati nam in fugit tempora ex numquam. Placeat fugiat odit quia qui, ex excepturi eius ut non sed.
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;
